testsuite/src/main/java/io/netty/testsuite/transport/socket/SocketConnectionAttemptTest.java

```diff
@@ -96,7 +95,7 @@ public void channelActive(ChannelHandlerContext ctx) throws Exception {
 
         cb.handler(handler);
         cb.option(ChannelOption.ALLOW_HALF_CLOSURE, halfClosure);
-        ChannelFuture future = cb.connect(NetUtil.LOCALHOST, TestUtils.getFreePort()).awaitUninterruptibly();
+        ChannelFuture future = cb.connect(NetUtil.LOCALHOST, BAD_PORT).awaitUninterruptibly();
```

Comment:
Do we really want to test connect is refused to a port that cannot be bound to ... or do we want to test that connection is refused to a port that could be bound to ... but that no-one is currently bound to?